ci/kokoro/docker/build-in-docker-bazel.sh

```diff
@@ -66,7 +66,7 @@ readonly BAZEL_CACHE="https://storage.googleapis.com/cloud-cpp-bazel-cache"
 # not hit this cache.
 if [[ -r "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
   io::log "Using bazel remote cache: ${BAZEL_CACHE}"
-  bazel_args+=("--remote_cache=${BAZEL_CACHE}")
+  bazel_args+=("--remote_cache=https://storage.googleapis.com/cloud-cpp-bazel-cache/docker/${BUILD_NAME}")
```

Comment:
> Just so I'm clear, this change is purely to give us the ability to clear this cache in the future _if_ we discover a problem w/ bazel's remote cache. We've currently not seen it cause any problems, nor do we have any need to clear this cache, is that right?

Effectively I cleared the cache by starting from a fresh one.  And yes, we are seeing problems.  What *exactly* are the problems I am not sure.


All I know is that the PR builds with both the tarball and the existing Bazel cache break, that removing the tarball fixed some builds but later they started failing again, and that clearing both caches seems to have fixed the builds for good.  I don't know, maybe the underlying Docker image used to create the tarball matters and Bazel does not include enough of it into the cache inputs.

I am not going to spend time troubleshooting Bazel, clearing the caches worked.

> 
> This will likely reduce our cache hit rate a little, but probably a negligible amount,

I think the amount will be zero: the Bazel builds all have different command-line options (*san vs. libcxx vs. Windows vs. macOS) all of which are caching inputs.  I think none of the artifacts created by build A are used in build B.  I guess maybe some of the `genrule()` artifacts are reusable... hmm... I am wrong it is not zero, but it is something like 1/5,000.